numpy/core/tests/test_deprecations.py

```diff
@@ -618,3 +618,25 @@ def test_not_deprecated(self):
             self.assert_not_deprecated(round, args=(scalar,))
             self.assert_not_deprecated(round, args=(scalar, 0))
             self.assert_not_deprecated(round, args=(scalar,), kwargs={'ndigits': 0})
+
+
+class TestIncorrectAdvancedIndexWithEmptyResult(_DeprecationTestCase):
+    # 2020-04-02, NumPy 1.19.0
+    message = "Out of bound index found. This was previously ignored"
+
+    @pytest.mark.parametrize("index", [([3, 0],), ([0, 0], [3, 0])])
+    def test_empty_subspace(self, index):
+        # Test for both a single and two/multiple advanced indices. These
+        # This will raise an IndexError in the future.
+        arr = np.ones((2, 2, 0))
+        self.assert_deprecated(arr.__getitem__, args=(index,))
+        self.assert_deprecated(arr.__setitem__, args=(index, []))
```

Comment:
Perhaps worth adding the test I allude to above:
```suggestion
        self.assert_deprecated(arr.__setitem__, args=(index, []))
        
        # for this array, the subspace is only empty after applying the slice
        arr2 = np.ones((1, 2, 2))
        index2 = (slice(None),) + index
        self.assert_deprecated(arr2.__getitem__, args=(index2,))
        self.assert_deprecated(arr2.__setitem__, args=(index2, []))
```

Reply:
Needed some fixups, but adapted... And mofied the code to not check indices if `NpyIter_GetSize(mit->outer) == 0`, which should bring it fully in line with what you expected. The error message/release notes is a bit adapted...

Reply:
Whoops, guess I forgot how slices work